clarity-ext-scripts/clarity_ext_scripts/covid/knm_sminet_service.py

```diff
@@ -0,0 +1,157 @@
+from clarity_ext.utils import lazyprop
+from sminet_client import (SampleInfo, ReferringClinic, Patient, Doctor, SmiNetConfig, NotificationType,
+                           LabDiagnosisType, LabResult, Laboratory, SmiNetLabExport)
+from sminet_client import Gender as SmiNetGender
+from clarity_ext_scripts.covid.knm_service import KNMConfig, ServiceRequestProvider
+
+
+class KNMSmiNetIntegrationService(object):
+
+    MAP_GENDER_FROM_KNM_TO_SMINET = {
+        "male": SmiNetGender.MALE,
+        "female": SmiNetGender.FEMALE,
+        "other": SmiNetGender.UNKNOWN,
+        "unknown": SmiNetGender.UNKNOWN,
+        None: SmiNetGender.UNKNOWN,
+    }
+
+    def __init__(self, config):
+        self.config = config
+
+        knm_config = KNMConfig(config)
+        sminet_config = SmiNetConfig(**config)
+
+        self.knm_client = PartnerAPIV7Client(**knm_config)
+        self.sminet_client = SmiNetClient(sminet_config)
+
+    def get_county_from_organization(self, organization):
+        """
+        Given the raw json from the KNM service, returns a county understood by SmiNet
+        """
+        aliases = organization["alias"]
+        for alias in aliases:
+            county_code = alias.split("-")[0]
+            # TODO: THIS IS JUST FOR TEMPORARY TEST PURPOSES (waiting for knm fix)
+            if county_code == "SE":
+                county_code = "AB"
+
+            if self.sminet_client.is_supported_county_code(county_code):
+                return county_code
+        raise NoSupportedCountyCodeFound(
+            "No supported county code found in alias list. Found: {}".format(aliases))
+
+    def create_sample_info(self, provider, sample):
+        sample_date_referral = provider.service_request["resource"]["authoredOn"]
+        # The date is on ISO8601 format:
+        sample_date_referral = dateutil.parser.isoparse(sample_date_referral)
+
+        return SampleInfo(status=1,
+                          sample_id=sample.org_referral_code,
+                          sample_date_arrival=sample.date_arrival,
+                          sample_date_referral=sample_date_referral,
+                          sample_material="Svalg",
+                          sample_free_text_referral="Anamnes: Personalprov")
+
+    def create_referring_clinic(self, provider):
+        """Creates a referring clinic object from KNM data"""
+        referring_clinic_name = provider.patient["managingOrganization"]["display"]
+        referring_clinic_county = self.get_county_from_organization(
+            provider.organization)
+        referring_doctor = provider.service_request["resource"]["requester"]["display"]
+
+        # TODO: We don't have an address for the referring clinic
+        return ReferringClinic(referring_clinic_name, "",
+                               referring_clinic_county, Doctor(referring_doctor))
+
+
+    def map_gender(self, knm_gender):
+        """Returns gender in the format required by SmiNet"""
+        pass
```

Comment:
Silent, so I would presume unused method?

Reply:
Yes, I'll fill it in with that code that uses the map.